### DISCUSSION TITLE
can build ServiceWorker decompress version?

### DISCUSSION THREAD

**Discussion by nenge123** [gildas-lormeau/zip.js#454] [Ideas]
can build sericeWorker decompress version?

the `new BlobReader` can auto?
```javascript
function initReader(reader) {
    if (Array.isArray(reader)) {
        reader = new SplitDataReader(reader);
    }
    if(reader instanceof Blob){
        reader = new BlobReader(reader);
    }
    if((reader.buffer||reader) instanceof arrayBuffer){
        reader = new Uint8ArrayReader(reader.buffer||reader);
    }
    if (reader instanceof ReadableStream) {
        reader = {
            readable: reader
        };
    }
    return reader;
}
```

**Comment by gildas-lormeau** [comment]
Could you tell me more precisely what you mean? You can find an example of service worker which uses zip.js to decompress files here: https://github.com/gildas-lormeau/zip.js/blob/1162729066581bfd78ee73cb6708e6754ba5cc3d/tests/all/test-sw-worker.js. The music player in the demo "Zip Manager" does something similar, see https://github.com/gildas-lormeau/zip-manager/blob/1653f0c1759b6b3afa0608f3a31fa39bab736fc8/src/sw.js#L60-L75.

**Comment by nenge123** [comment]
I know why my error.it's lost `getData(new zip.BlobWriter());`

```
await (await new zip.ZipReader(new zip.BlobReader(await (await fetch('z.zip')).blob())).getEntries())[0].getData();
```
zip-full.js:7279 Uncaught TypeError: Cannot read properties of undefined (reading 'writable')
    at initWriter (zip-full.js:7278:14)
    at ZipEntry.getData (zip-full.js:7779:13)
    at async <anonymous>:1:5
